vllm/transformers_utils/tokenizers/mistral.py

```diff
@@ -165,18 +165,20 @@ def apply_chat_template(self,
                             messages: List["ChatCompletionMessageParam"],
                             tools: Optional[Dict[str, Any]] = None,
                             **kwargs) -> List[int]:
-        assert tools is None, "`tools` are not yet supported."
 
-        request = ChatCompletionRequest(
-            messages=messages)  # type: ignore[type-var]
+        request = ChatCompletionRequest(messages=messages,
+                                        tools=tools)  # type: ignore[type-var]
         encoded = self.mistral.encode_chat_completion(request)
 
         # encode-decode to get clean prompt
         return encoded.tokens
 
     def convert_tokens_to_string(self, tokens: List[str]) -> str:
         if isinstance(self.tokenizer, Tekkenizer):
-            return "".join(tokens)
+            return "".join([
+                t for t in tokens
+                if t not in self.tokenizer._all_special_tokens
+            ])
```

Comment:
```suggestion
            return "".join(
                t for t in tokens
                if t not in self.tokenizer._all_special_tokens
            )
```

`str.join` works on iterables so we can avoid iterating through it twice.